extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/ResteasyBuiltinsProcessor.java

```diff
@@ -75,4 +87,49 @@ void vertxProviders(BuildProducer<ResteasyJaxrsProviderBuildItem> providers) {
         providers.produce(new ResteasyJaxrsProviderBuildItem(JsonObjectReader.class.getName()));
         providers.produce(new ResteasyJaxrsProviderBuildItem(JsonObjectWriter.class.getName()));
     }
+
+    private static void transformEagerSecurityNativeMethod(
+            BuildProducer<BytecodeTransformerBuildItem> bytecodeTransformerProducer) {
+
+        // 1. add getResourceFactory() to ResourceMethodInvoker
```

Comment:
RESTEasy doesn't provide any access to actually invoked class. By my experiments, most accurate results are received when scannable class (which is the lowest one, the non-abstract class) detected by Quarkus itself is inspected for that method and we use `getDeclaringClass` of that method (which is often not the scannable one). The scannable class is only available on `ResourceFactory` which is public. But access to ResourceFactory has protected visibility. So this creates a getter method.

I wrote to @jamezp so that we avoid this in the future, but IMO it's not an issue to add one getter for now. If we had unlimited time, I'd wait for the discussion and release of the RESTEasy, but there has been users asking for fix of #39964 and I don't see an issue here.

Reply:
@michalvavrik OK, got it, thanks for the clarification. Sure, hope James @jamezp can get a chance to make it directly available at the RESTEasy level directly at some point, and then we can simplify a bit, thanks

Reply:
I've asked for some clarification on the email on what is actually needed. There might be some other workarounds, but I'm also not opposed to adding a getter if I'm understanding the issue correctly.